### PR TITLE
feat: Implement llm system evaluation

### PR DESCRIPTION
**Pull request by Yuan325**
Implementation of llm system evaluation. 

Usage example: 

If would like to output detailed metric table (row-specific scores), add the following env var `export EXPORT_CSV=True`

Run the following: `python run_evaluation.py`

### CODE REVIEW

File: llm_demo/evaluation.cloudbuild.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+serviceAccount: "projects/$PROJECT_ID/serviceAccounts/evaluation-testing@retrieval-app-testing.iam.gserviceaccount.com" # Necessary for ID token creation
+options:
+  logging: CLOUD_LOGGING_ONLY # Necessary for custom service account
+  dynamic_substitutions: true
+
+substitutions:
+  _ORCHESTRATION_TYPE: "langchain-tools"
+  _RETRIEVAL_EXPERIMENT_NAME: "retrieval-phase-eval-${_PR_NUMBER}"
+  _RESPONSE_EXPERIMENT_NAME:  "response-phase-eval-${_PR_NUMBER}"
+
```
Comment (kurtisvg) [inline llm_demo/evaluation.cloudbuild.yaml:43]: Do these still get run on merge to main?

File: llm_demo/evaluation/eval_golden.py [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+# limitations under the License.
+
+from datetime import datetime, timedelta
+from typing import Any, Dict, List, Optional
+
+from pydantic import BaseModel, Field
+from pytz import timezone
+
+
+class ToolCall(BaseModel):
```
Comment (kurtisvg) [inline llm_demo/evaluation/eval_golden.py:22]: nit: some comments on these classes would be nice to have.

What is it representing (or doing)?

**Review by kurtisvg — APPROVED** [review]

File: llm_demo/evaluation.cloudbuild.yaml (hunk 1 above)
Comment (Yuan325) [reply llm_demo/evaluation.cloudbuild.yaml:43]: Nope, this cloudbuild.yaml only runs on PRs. if we want to run it periodically on main, we can set up a cloud build a replace the name for this.

**Pull request merged**